packages/frontend/component/src/components/share-menu/use-share-url.ts

```diff
@@ -5,29 +5,40 @@ import { toast } from '../../ui/toast';
 
 type UrlType = 'share' | 'workspace';
 
+export type ShareMode = 'page' | 'edgeless';
+
 type UseSharingUrl = {
   workspaceId: string;
   pageId: string;
   urlType: UrlType;
+  mode?: ShareMode;
 };
 
 export const generateUrl = ({
   workspaceId,
   pageId,
   urlType,
+  mode = 'page',
```

Comment:
if the mode has been saved in the DB, the value in the query seems unnecessary.

Reply:
the feature could redesign with mode data in the DB

Reply:
Do you mean removing the query when generating a link? Or does the link still carry the query, but the mode is returned by the database, hence no longer comes from an external input?